cppwinrt/code_writers.h

```diff
@@ -1308,6 +1308,18 @@ namespace cppwinrt
             static_cast<D const&>(*this).template as<IBufferByteAccess>()->Buffer(&data);
             return data;
         }
+)");
+        }
+        else if (type_name == "Windows.Foundation.IMemoryBufferReference")
```

Comment:
Should this return both the length and the data? I guess a caller would use `::Capacity()` but that costs another call.

Reply:
This would violate the convention of `data()` returning only the pointer.

I think a better idea would be adding an implicit conversion to `std::span`, if someone wants to retrieve both at once they can then use this.

Reply:
The problem with an implicit conversion is that if you only have a single reference to the object and then assign it directly from a temporary then the returned std::span would be invalid and you'd not notice. Instead, it's better to have a method named as_span.

Reply:
`as_span()` wouldn't make that detail immediately obvious either.

Reply:
I'm not a C++ programmer, so I don't have an opinion here about spans. Having a `data()` method is continent for the WinRT Python bindings since `IBuffer` already has the same.

Reply:
Unfortunately, `IMemoryBufferByteAccess` and `IBufferByteAccess` are quite different - in hindsight the same interface should probably have been used for both classes. The original `data` method was modeled after `std::vector`'s `data` method, providing "direct access to the underlying array". Let's stick with that as that is a well-known convention. We can add explicit conversions to `std::span` to both later on if that ends up being valuable. That way they remain largely consistent.

Reply:
I'd prefer not to add more uses of array_view as it should be "deprecated" in favor of std::span - the trouble is span is new to C++20 and we haven't yet made the break with C++17.

Reply:
That's fair. We can keep this in our back pocket once we break with C++17.

Reply:
On the other hand, I just remembered that we've already been using [feature testing](https://en.cppreference.com/w/cpp/feature_test) to enable C++20 features for coroutines and ranges. We can just do the same for std::span.